### PR TITLE
Separate article header and article content configuration in content profiles (SDESK 3409)

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDESK-3409

### CODE REVIEW

**Comment by tomaskikutis** [comment]
Ignore the failing test, it's not related to this PR. It's an issue of API returning desk members in random order. Will be fixed in SDESK-3566

File: scripts/core/helpers/typescript-helpers.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export function nameof<T>(name: keyof T) {
     return name;
 }
+
+export function assertNever(x: never): never {
```
Comment (pablopunk) [inline scripts/core/helpers/typescript-helpers.ts]: what is this function for?

Comment (tomaskikutis) [reply scripts/core/helpers/typescript-helpers.ts]: It makes exhaustive switch possible.

```
type sectionToRender = 'header' | 'content';

function handleIt (section: sectionToRender) {
    switch(section) {
        case 'header':
            // ...
            break;
        case 'content':
            // ...
            break;
        default:
            assertNever(section)
    }
}
```

If you add another possible value to `sectionToRender` type, the code won't compile until you handle it in all switch statements which switch on that type.

Comment (pablopunk) [reply scripts/core/helpers/typescript-helpers.ts]: Cool

**Pull request merged**